### PR TITLE
[FIX] pyramid_chart: hide zero values and improve label positioning for small bars

### PR DESCRIPTION
**Pull request by dhrp-odoo**
## Description

This PR addresses two issues related to the `showValues` plugin in pyramid charts:

### 1. Incorrect Display of Negative Values

The absolute value display for pyramid chart labels was already handled in PR #6535
This PR adds a missing test to validate that behavior and ensures zero values are not displayed, improving clarity.

### 2. Label Positioning for Small Bars

Enhance value label positioning logic for the following chart types:

- Horizontal bar
- Vertical bar
- Pyramid
- Line

Labels are now anchored near the zero line when the bar size is too small to fit the text.
This avoids overlap, ensures better spacing, and improves readability across various chart types.

image

Task: [4762696](https://www.odoo.com/odoo/2328/tasks/4762696)

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_t("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

### CODE REVIEW

File: src/helpers/figures/charts/runtime/chartjs_show_values.ts [hunk 1]
```diff
@@ -7,10 +7,19 @@ export function getChartShowValues(
   args: ChartRuntimeGenerationArgs
 ): ChartShowValuesPluginOptions {
   const { axisFormats, locale } = args;
+  const formatDatasetValue = formatChartDatasetValue(axisFormats, locale);
+  const isPyramidChart = definition.type === "pyramid";
```
Comment (hokolomopo) [inline src/helpers/figures/charts/runtime/chartjs_show_values.ts]: I would create a new function `getPyramidChartShowValues` instead, as we do for the other helpers (scale/legend/...)

Also it might need an adaptation in odoo. Not sure if `odoo_pyramid_chart` already exists in 18.1 tho

File: src/helpers/figures/charts/runtime/chartjs_show_values.ts [hunk 2]
```diff
@@ -34,7 +34,9 @@ export function getPyramidChartShowValues(
     background: definition.background,
```
Comment (hokolomopo) [inline src/helpers/figures/charts/runtime/chartjs_show_values.ts:34]: nitpick: commit title is "[FIX] pyramid_chart: showValues plugin displays absolute values", but the thing you fix here is the display of the zeros.

File: tests/figures/chart/pyramid_chart/pyramid_chart_plugin.test.ts [hunk 3]
```diff
@@ -124,5 +125,23 @@ describe("population pyramid chart", () => {
       expect(options?.scales?.x?.suggestedMin).toBe(-33);
       expect(options?.scales?.x?.suggestedMax).toBe(33);
     });
+
+    test("pyramid chart showValues plugin do not show negative values", () => {
```
Comment (hokolomopo) [inline tests/figures/chart/pyramid_chart/pyramid_chart_plugin.test.ts]:
```suggestion
    test("pyramid chart showValues plugin do not show negative or zeroes values", () => {
```
nitpick

File: src/components/figures/chart/chartJs/chartjs_show_values_plugin.ts [hunk 4]
```diff
@@ -134,18 +136,27 @@ function drawHorizontalBarChartValues(
       const point = dataset.data[i];
 
       const yPosition = point.y;
-      let xPosition = value < 0 ? point.x + point.width / 2 : point.x - point.width / 2;
-      xPosition = Math.min(xPosition, xMax);
-      xPosition = Math.max(xPosition, xMin);
+      const textWidth = computeTextWidth(ctx, displayValue, { fontSize: 12 }, "px");
+      const barWidth = Math.abs(point.x - xZero);
```
Comment (hokolomopo) [inline src/components/figures/chart/chartJs/chartjs_show_values_plugin.ts]:
```suggestion
      const distanceFromAxisOrigin = Math.abs(point.x - xZero);
```
or something like that. "barWidth" is a bit confusing for stacked charts, we'd expect that it'd be the width of the bar, which is not the case

File: src/components/figures/chart/chartJs/chartjs_show_values_plugin.ts [hunk 5]
```diff
@@ -134,18 +136,27 @@ function drawHorizontalBarChartValues(
...
 
       // Avoid overlapping texts with same Y
       if (!textsPositions[yPosition]) {
         textsPositions[yPosition] = [];
       }
-      const textWidth = computeTextWidth(ctx, displayValue, { fontSize: 12 }, "px");
       for (const otherPosition of textsPositions[yPosition]) {
         if (Math.abs(otherPosition - xPosition) < textWidth) {
-          xPosition = otherPosition + textWidth + 3;
+          xPosition =
```
Comment (hokolomopo) [inline src/components/figures/chart/chartJs/chartjs_show_values_plugin.ts:169]: the same issue exists for vertical stacked bar charts. Maybe we could also fix it here ?

**Review by hokolomopo — COMMENTED** [review]
:wave: hello hello

File: src/helpers/figures/charts/runtime/chartjs_show_values.ts (hunk 2 above)
Comment (dhrp-odoo) [reply src/helpers/figures/charts/runtime/chartjs_show_values.ts:34]: Yup — since the original task was to fix the display of negative values, but since that's now being handled in your [PR](https://github.com/odoo/o-spreadsheet/pull/6535), I'll update the commit message accordingly. 👍

File: src/components/figures/chart/chartJs/chartjs_show_values_plugin.ts (hunk 5 above)
Comment (dhrp-odoo) [reply src/components/figures/chart/chartJs/chartjs_show_values_plugin.ts:169]: ✅ Done for Line, Bar (both vertical and horizontal) charts.

**Comment by hokolomopo** [comment]
@robodoo r+

**Comment by hokolomopo** [comment]
@robodoo rebase-ff
